packages/dev/core/src/Shaders/screenSpaceReflection.fragment.fx

```diff
@@ -174,3 +196,338 @@ void main()
+            info.visibilityBackup = 1.0;
+        #endif
+
+    } else {
+        info.miss = false;
+        vec3 normalInUV = normalize(texture2D(normalSampler, uv).xyz);
+        if (dot(normalize(dirVS), normalInUV) > 0.0){
+            info.visibility = 0.0; // hit backface of a mesh
+        } else {
+            vec2 dCoordScreen = smoothstep(vec2(0.2), vec2(0.6), abs(vec2(0.5, 0.5) - uv)); // HermiteInterpolation
+            info.visibility = texture2D(textureSampler, uv).a // alpha value of the reflected scene position 
+                * abs(dot(normalize(hitCoordVS), normalInUV)) // to avoid artifacts due to missing information when reflected face is almost perpendicular to the camera view direction
+        
+                // reduce back face limit artifacts (false positive intersections)
+                * clamp(dot(normalInUV, normalize( texture2D(normalSampler, (currFrag + increment * resolution)/texSize).xyz)), 0.0, 1.0)
+                * clamp(dot(normalInUV, normalize( texture2D(normalSampler, (currFrag - increment * resolution)/texSize).xyz)), 0.0, 1.0)
+                * clamp(dot(normalInUV, normalize( texture2D(normalSampler, (currFrag + increment.yx * resolution)/texSize).xyz)), 0.0, 1.0)
+                * clamp(dot(normalInUV, normalize( texture2D(normalSampler, (currFrag - increment.yx * resolution)/texSize).xyz)), 0.0, 1.0)
+                
+                * (1.0 - max (dot(-normalize(hitCoordVS), normalize(dirVS)), 0.0)) // to fade out the reflection as the reflected direction point to the camera's position (hit behind the camera)
+                * (1.0 - currSearch) // the reflection should be sharper when near from the starting point
+                * (1.0 - clamp (abs(hitCoordVS.z / distanceFade), 0.0, 1.0)) // to fade out the reflection near the distanceFade
+                * (1.0 - clamp (deltaDepth/maxTol, 0.0, 1.0)) // since the hit point is not always precisely found, we fade out the reflected color if we aren't precise enough 
+                * clamp(1.0 - (dCoordScreen.x + dCoordScreen.y), 0.0, 1.0); // to fade out the reflection near the edge of the screen
+        }
+
+        info.visibility = clamp(info.visibility * 2.0, 0.0, 1.0); // to compensate for previous lines to have strong enough reflections
+
+        #if defined(BACKUP_TEXTURE_SKYBOX) || defined(BACKUP_TEXTURE_PROBE)
+            info.visibilityBackup = 1.0 - info.visibility; // complementary reflectivityColor
+        #endif
+        info.coords = uv;
+    }
+
+    return info;
+}
+
+void main(void)
+{
+    #ifdef SSR_SUPPORTED
+
+    // *************** Get data from samplers ***************
+
+    vec4 original = texture2D(textureSampler, vUV);
+    vec3 spec = toLinearSpace(texture2D(reflectivitySampler, vUV).rgb);
+
+    if (dot(spec, vec3(1.0)) <= 0.0){
+        gl_FragColor = original; // no reflectivity, no need to compute reflection
+        return;
+    }
+
+    float roughness = 1.0 - texture2D(reflectivitySampler, vUV).a;

```

Comment:
You should set `vec4 spec = toLinearSpace(texture2D(reflectivitySampler, vUV));` and do `roughness = 1.0 - spec.a` to save a texture lookup (`toLinearSpace` won't change the `alpha` component).